plugin/src/main/java/org/opennms/velocloud/shell/ListPartnerConnectionsCommand.java

```diff
@@ -0,0 +1,92 @@
+ * by the Free Software Foundation, either version 3 of the License,
+ * or (at your option) any later version.
+ *
+ * OpenNMS(R) is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with OpenNMS(R).  If not, see:
+ *      http://www.gnu.org/licenses/
+ *
+ * For more information contact:
+ *     OpenNMS(R) Licensing <license@opennms.org>
+ *     http://www.opennms.org/
+ *     http://www.opennms.com/
+ *******************************************************************************/
+package org.opennms.velocloud.shell;
+
+import com.google.common.base.Strings;
+import org.apache.karaf.shell.api.action.Action;
+import org.apache.karaf.shell.api.action.Argument;
+import org.apache.karaf.shell.api.action.Command;
+import org.apache.karaf.shell.api.action.Completion;
+import org.apache.karaf.shell.api.action.lifecycle.Reference;
+import org.apache.karaf.shell.api.action.lifecycle.Service;
+import org.apache.karaf.shell.api.console.Session;
+import org.apache.karaf.shell.support.table.Col;
+import org.apache.karaf.shell.support.table.ShellTable;
+import org.opennms.velocloud.client.api.VelocloudApiClientProvider;
+import org.opennms.velocloud.client.api.VelocloudApiException;
+import org.opennms.velocloud.connections.Connection;
+import org.opennms.velocloud.connections.ConnectionManager;
+
+import java.util.Optional;
+
+@Command(scope = "opennms-velocloud", name = "list-msp-partner-connections", description = "List MSP Connections", detailedDescription = "List Connections for MSP")
+@Service
+public class ListPartnerConnectionsCommand implements Action {
+
+    @Reference
+    public ConnectionManager connectionManager;
+
+    @Reference
+    private Session session;
+
+    @Argument(index = 0, name = "alias", description = "Velocloud URL", required = true, multiValued = false)
+    @Completion(AliasCompleter.class)
+    public String alias = null;
+
+    @Override
+    public Object execute() throws Exception {
+
+        final var client = this.connectionManager.getPartnerClient(this.alias);
+        if (client.isEmpty()) {
+            System.err.println("No connection with alias " + this.alias);
+            return null;
+        }
+
+        final var table = new ShellTable()
+                .size(session.getTerminal().getWidth() - 1)
+                .column(new Col("ID").maxSize(12).bold(true))
+                .column(new Col("Username").maxSize(24))
+                .column(new Col("Domain").maxSize(24))
+                .column(new Col("Email").maxSize(24))
+                .column(new Col("Role").maxSize(24))
+                .column(new Col("Active"));
+
+        for (final var user : client.get().getUsers()) {
```

Comment:
> Same confusion here, are connections between a partner its customers considered 'users'?

What I understood is partner connections are partner users, but thats my understanding, I might be wrong there too.

Reply:
This PR seems to be stalled - can we get it moving again please?

Reply:
I believe @amay94  has done a PR that is a mix of 3 jiras (including this one)  https://github.com/OpenNMS/opennms-velocloud-plugin/pull/29
There have been a lot of changes since this PR was started so if I'm correct this PR should be closed right @amay94 .

Reply:
I believe so, although the work I did just listed the connections that have been configured within OpenNMS to connect to orchestrators. This retrieves a list of connections from an MSP partner to it's customers, which looks similar to what [NMS-14516](https://issues.opennms.org/browse/NMS-14516) is/was asking for.

Reply:
Closing since this requirement was misinterpreted.